src/traffic_layout/engine.cc

```diff
@@ -566,6 +332,99 @@ RunrootEngine::copy_runroot(const std::string &original_root, const std::string
   } else {
     path_map[LAYOUT_PREFIX] = ".";
   }
+
+  YAML::Emitter yamlfile;
+  // emit key value pairs to the yaml file
+  yamlfile << YAML::BeginMap;
+  for (uint i = 0; i < dir_vector.size(); i++) {
+    yamlfile << YAML::Key << dir_vector[i];
+    yamlfile << YAML::Value << path_map[dir_vector[i]];
+  }
+  yamlfile << YAML::EndMap;
+
+  // create the file
+  std::ofstream f(Layout::relative_to(ts_runroot, "runroot.yaml"));
+  f << yamlfile.c_str();
+}
+
+// check if user want to force remove the ts_runroot
+// return true if user replies Y
+static bool
+check_remove_force()
+{
+  // check for Y/N 3 times
+  for (int i = 0; i < 3; i++) {
+    std::cout << "Are you sure to force removing runroot? (irreversible) Y/N: ";
+    std::string input;
+    std::cin >> input;
+    if (input == "Y" || input == "y")
+      return true;
+    if (input == "N" || input == "n")
+      return false;
+  }
+  ink_error("Invalid input Y/N");
+  exit(70);
+}
+
+// the function for removing the runroot
+void
+LayoutEngine::remove_runroot()
+{
+  std::string path = path_handler(arguments.get("path").value(), false, _argv[0]);
+  // check validity of the path
+  if (path.empty()) {
+    ink_fatal("Path not valid (runroot.yaml not found)");
```

Comment:
Isn't that more "path not provided"?

Reply:
The path_handler returns an empty string if the `runroot.yaml` is not found even if we give provide a path. It should be `not valid`.